src/char/int_party.c

```diff
@@ -72,55 +72,74 @@ static int inter_party_check_lv(struct party_data *p)
 
 	return 1;
 }
-//Calculates the state of a party.
+
+/**
+ * Checks if a party is a family party.
+ *
+ * @param p The party.
+ * @return The child's char ID on success, otherwise 0.
+ *
+ */
+static int inter_party_is_family_party(struct party_data *p)
+{
+	nullpo_ret(p);
+
+	if (p->size < 2 || p->size > 3)
+		return 0;
+
+	for (int i = 0; i < MAX_PARTY - 1; i++) {
+		if (p->party.member[i].online == 0)
+			continue;
+
+		struct mmo_charstatus *char_i = (struct mmo_charstatus*) idb_get(chr->char_db_, p->party.member[i].char_id);
+
+		if (char_i == NULL)
+			continue;
+
+		for (int j = i + 1; j < MAX_PARTY; j++) {
+			if (p->party.member[j].online == 0)
+				continue;
+
+			struct mmo_charstatus *char_j = (struct mmo_charstatus*) idb_get(chr->char_db_, p->party.member[j].char_id);
+
+			if (char_j == NULL)
+				continue;
+
+			if (char_i->char_id == char_j->child)
+				return char_i->char_id;
```

Comment:
It this supposed to return successfully if there's a stranger online in the party?

For example:

- Parent
- Child
- Stranger

Reply:
You're right, This case was not checked. Thanks. :)